Server/src/main/java/net/minecrell/serverlistplus/server/ServerListPlusServer.java

```diff
@@ -238,7 +224,7 @@ public int getProtocolVersion() {
         // Favicon
         FaviconSource favicon = response.getFavicon();
         if (favicon != null && favicon != FaviconSource.NONE) {
-            Optional<String> icon = faviconCache.getUnchecked(favicon);
+            Optional<String> icon = faviconCache.get(favicon).toJavaUtil();
             icon.ifPresent(ping::setFavicon);
```

Comment:
Seems a bit overkill to convert to `java.util.Optional` just to do the weird ifPresent here (I never really liked those to be honest, looks confusing). Can you change this to `if (icon.isPresent()) ping.setFavicon(icon.get()));`?

Reply:
Sure no problem!

Reply:
I left in the conversion to Java's `Optional` because Java's Optional is used in other places in `SpongePlugin` and `VelocityPlugin`. I could remove the conversion from `ServerListPlusServer`, but `OptionalInt` is used in that class, so I left it in for consistency.

Reply:
If you don't mind having a fully-qualified class name, I can remove the conversion.